pkg/snapstore/gcs_snapstore.go

```diff
@@ -96,6 +103,21 @@ func NewGCSSnapStore(config *brtypes.SnapstoreConfig) (*GCSSnapStore, error) {
 	return NewGCSSnapStoreFromClient(config.Container, config.Prefix, config.TempDir, config.MaxParallelChunkUploads, config.MinChunkSize, chunkDirSuffix, gcsClient), nil
 }
 
+func getGCSStorageAPIEndpoint(path string) (string, error) {
+	if _, err := os.Stat(path); err == nil {
+		data, err := os.ReadFile(path)
+		if err != nil {
+			return "", err
+		}
+		if len(data) != 0 {
```

Comment:
If the file is of size 0, etcd-backup-restore rejects an explicit user configuration, and falls back to reading from the env variable.
Notifying the user of the misconfiguration instead of falling back to reading from the environment variable, which also might not have the necessary endpoint; would be more convinient from a user's perspective (in my opinion).
If the file size is indeed 0, and the env var is unset, the default endpoint is used without the user coming to know that the override has failed.
Thoughts?

Reply:
Thanks for the suggestion. I've included this check and now return error upon empty file. So if the file is empty, it does not fall back to default endpoint, but rather returns an error.

But your point about user misconfiguration can still happen, since we anyway can't validate the exact endpoint string provided. So if the user provides the endpoint as `https://googleappis.com` (with a typo), then anyway they'll get an error that the endpoint is invalid, so I think this case should still be ok.

Reply:
Thanks!